Scripts/post.js

```diff
@@ -0,0 +1,298 @@
+// post.js
+import { fetchPostById } from "./api.js";
+import { API_POSTS_URL, API_BASE_URL } from "./common.js";
+
+// the page for single posts:
+async function displaySinglePost() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const postId = urlParams.get("postId");
+  if (postId) {
+    try {
+      const post = await fetchPostById(postId);
+      renderPost(post);
+    } catch (error) {
+      console.error("Error fetching post:", error);
```

Comment:
No point only logging the error, display a message to the user using the displayErrorMessage function.